presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -952,6 +952,24 @@ public static long timeZoneHourFromTimestampWithTimeZone(@SqlType(StandardTypes.
         return extractZoneOffsetMinutes(timestampWithTimeZone) / 60;
     }
 
+    @Description("time zone hour of the given timestamp")
+    @ScalarFunction("last_day")
+    @SqlType(StandardTypes.DATE)
+    public static long lastDayOfMonthFromTimestamp(ConnectorSession session, @SqlType(StandardTypes.TIMESTAMP) long timestamp)
+    {
+        DateTime dateTime = new DateTime(getChronology(session.getTimeZoneKey()).millis().getMillis(timestamp));
+        return MILLISECONDS.toDays(dateTime.dayOfMonth().withMaximumValue().getMillis());
+    }
+
+    @Description("time zone hour of the given date")
+    @ScalarFunction("last_day")
+    @SqlType(StandardTypes.DATE)
+    public static long lastDayOfMonthFromDate(@SqlType(StandardTypes.DATE) long date)
```

Comment:
Maybe we do not need this method. The `lastDayOfMonthFromTimestamp` should be enough. And if given a `DATE` argument an automatic coercion `DATE -> TIMESTAMP` will be used.